src/log.c

```diff
@@ -81,3 +83,46 @@ void dprint_data(const void *data, size_t datasz)
     }
     DPRINTF("]\n");
 }
+
+/*
+ * Initialize logging.
+ *
+ * Sets the loglevel based on the config variable "loglevel" stored
+ * in /etc/uefistored/uefistored.conf.
+ */
+void logging_init(void)
+{
+    FILE *fd;
+    char *buf = NULL;
+    size_t sz = 0;
+    char *name = NULL;
+    char *val = NULL;
+    char *ptr = NULL;
+
+    fd = fopen("/etc/uefistored/uefistored.conf", "r");
+    if (!fd) {
+        INFO("No /etc/uefistored/uefistored.conf found.\n");
+        return;
+    }
+
+    while (getline(&buf, &sz, fd) != -1) {
```

Comment:
@Wescoeur do we already have utility functions to parse configuration files somewhere? It feels backwards to do such parsing manually.

Reply:
We don't have functions to do that but this parsing is not very complex so I suppose it's fine to keep it unmodified.

Reply:
Ok. I don't feel so good about such code because there exist more robust and flexible implementations for reading configuration files, allowing for comments, tolerant to extra spaces, etc. But I don't know if such libs are easily available without adding extra deps.

Reply:
I don't like it either, which is why I looked around for a GPLv2 licensed, single file option... but the projects I was seeing were either of a different license, in C++, or were so big it felt like killing an ant with a bazooka.

Ideally it would be one header and one source file, and then calling it out separately in LICENSE, no?

I'd almost rather we change the file to be `/etc/uefistored/debug` and it contains a single character `0` or `1` t(0 is the default INFO, 1 is DEBUG) than bring in another unnecessarily large code base.

Let me poke around again for more parsers before we make any further decisions...

Reply:
... I'll have something soon.  Spent a few hours and ripped the `busybox` parser out of it.  Will push it today or tomorrow.

Reply:
Actually I'm not fond either of embedding third party libs directly into our projects, in general. Well, I'm not entirely against in this particular case if there's no better way, but to me the best way would be to use a well known lib that would be widely used enough that it is available on all linux distros. In the case of dom0, I think there probably is such lib available already given the number of programs written in C that need to manipulate configuration files. For example, this answer suggests glib's key-value-file-parser, or libconfig (which is available in dom0 too): https://stackoverflow.com/questions/147902/linux-configuration-file-libraries. If there isn't any available lib, then I would consider embedding to avoid adding an extra dependency.

Embedded libs are frowned upon by packagers, who often spend time unembedding them in order to use the shared libs from the distro: the main goal is security: if the lib has a security flaw, you don't have to chase it down in every project that embeds it. You just fix it once.

Reply:
I'll try the GLib routine.  I'm disappointed I didn't find this in my search.